module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -540,9 +540,17 @@ namespace app
             }
             if (!isCurrentWindow(msg->getWindowName())) {
                 if (!windowsStack.isEmpty()) {
-                    const auto closeReason = msg->getReason() == SwitchReason::PhoneLock
-                                                 ? gui::Window::CloseReason::PhoneLock
-                                                 : gui::Window::CloseReason::WindowSwitch;
+                    auto closeReason = gui::Window::CloseReason::WindowSwitch;
```

Comment:
just saying this code is messed up. This callback is called on switch on previous window, not exactly on close (and we can miss on close too - I've added issue for that today :) ) 
saying that - it's ok.